Content.Server/Explosions/ExplosionHelper.cs

```diff
@@ -26,129 +32,162 @@ public static class ExplosionHelper
-            var entitiesAll = serverEntityManager.GetEntitiesInRange(coords, maxRange).ToList();
+            var exAct = entitySystemManager.GetEntitySystem<ActSystem>();
+            
+            var entitiesInRange = serverEntityManager.GetEntitiesInRange(mapId, boundingBox, 0).ToList();
 
-            foreach (var entity in entitiesAll)
+            foreach (var entity in entitiesInRange)
             {
-                if (entity.Deleted)
-                    continue;
-                if (!entity.Transform.IsMapTransform)
-                    continue;
-
-                if (!entity.Transform.Coordinates.TryDistance(entityManager, coords, out var distance))
+                if (entity.Deleted || !entity.Transform.IsMapTransform)
                 {
                     continue;
                 }
 
-                ExplosionSeverity severity;
-                if (distance < devastationRange)
+                if (!entity.Transform.Coordinates.TryDistance(entityManager, epicenter, out var distance) || distance > maxRange)
                 {
-                    severity = ExplosionSeverity.Destruction;
+                    continue;
                 }
-                else if (distance < heavyImpactRange)
+
+                var severity = CalculateSeverity(distance, devastationRange, heaveyRange);
+                exAct.HandleExplosion(epicenter, entity, severity);
+            }
+        }
+
+        /// <summary>
+        /// Damage tiles inside the range. The type of tile can change depending on a discrete
+        /// damage bracket [light, heavy, devastation], the distance from the epicenter and
+        /// a probabilty bracket [<see cref="LightBreakChance"/>, <see cref="HeavyBreakChance"/>, 1.0].
+        /// </summary>
+        ///
+        private static void DamageTilesInRange(EntityCoordinates epicenter,
+                                               GridId gridId,
+                                               Box2 boundingBox,
+                                               float devastationRange,
+                                               float heaveyRange,
+                                               float maxRange)
+        {
+            var mapManager = IoCManager.Resolve<IMapManager>();
+            if (!mapManager.TryGetGrid(gridId, out var mapGrid))
+            {
+                return;
+            }
+
+            var entityManager = IoCManager.Resolve<IEntityManager>();
+            if (!entityManager.TryGetEntity(mapGrid.GridEntityId, out var grid))
+            {
+                return;
+            }
+
+            _ = grid.TryGetComponent(out GridAtmosphereComponent atmosphereComponent);
+
+            var robustRandom = IoCManager.Resolve<IRobustRandom>();
+            var tileDefinitionManager = IoCManager.Resolve<ITileDefinitionManager>();
+
+            var tilesInGridAndCircle = mapGrid.GetTilesIntersecting(boundingBox);
+
+            foreach (var tile in tilesInGridAndCircle)
```

Comment:
This could use an InRangeUnobstructed check from the epicenter to the center of the tile, just so explosions don't affect tiles it can't reach (for example, tiles behind walls or other obstructions)